rc/lsp.kak

```diff
@@ -8,12 +8,19 @@ declare-option -docstring "Command with which lsp is run" str lsp_cmd "kak-lsp -
 
 # Faces used by inline diagnostics.
 set-face global DiagnosticError red
+set-face global DiagnosticHint cyan
+set-face global DiagnosticInfo blue
 set-face global DiagnosticWarning yellow
 # Faces used by inlay diagnostics.
 set-face global InlayDiagnosticError DiagnosticError
+set-face global InlayDiagnosticHint DiagnosticHint
+set-face global InlayDiagnosticInfo DiagnosticInfo
 set-face global InlayDiagnosticWarning DiagnosticWarning
-# Line flags for errors and warnings both use this face.
-set-face global LineFlagErrors red
+# Faces used by line flags
+set-face global LineFlagError red
+set-face global LineFlagHint cyan
+set-face global LineFlagInfo blue
+set-face global LineFlagWarning yellow
```

Comment:
I'm not sure if the "info" and "hint" faces should be coloured by default. After all, Kakoune doesn't even have a `Warning` face.

Not everyone wants their editor to pop in every colour of the rainbow, and these severities aren't as ubiquitous as something like "error == red".

Reply:
I think avoiding ambiguity is probably more important than avoiding color vomit. LSP servers are going to throw out any of the four diagnostic levels, and being able to know which one it is is pretty important. I will change hints to a dim white though, so it'll decrease the color pop a bit.

Reply:
It's definitely important for errors and warnings. And uncoloured would make "info" and "hint" visually different from the other two already. But I don't think the difference between those two is big enough that more colours are necessary.
I'd be willing to bet that even most servers don't have a clear distinction whether a message goes into "info" or "hint".

Reply:
Fair enough, I'll change it then. I'm going to keep them colored in my personal colorscheme, but for defaults, you're probably correct.

Reply:
![image](https://user-images.githubusercontent.com/3515394/132566898-d97813a6-d3db-4b44-9860-06bd9e29d12f.png)

Done (with old inlay diagnostics)

Reply:
I agree that it's safer to go with the default face.
If we can find some colors that work with the default colorschemes for most languages, then we can of course add them.

blue/cyan would be nice but Python uses blue for keywords, and Lua uses cyan for function names. It's a bit confusing if (unrelated) gutter chars have the same color.

It looks like the lua language server's info/hint diagnostics are like warnings, so in that case yellow would be fine.

Reply:
using yellow (at least for the gutter text) would also have that advantage that it's obvious that the gutter text is not part of the file

Reply:
Ok I somehow deleted my earlier commit because it appeard twice in GitHub..

Anyway, I just realized that the c_cpp filetype uses yellow for types, that kind of sucks..
I noticed rust-analyzer gives hints like "prefix variable with underscore" and so on, so yellow/warning wouldn't really here